jerry-core/ecma/builtin-objects/ecma-builtin-global.c

```diff
@@ -113,7 +113,7 @@ ecma_builtin_global_object_print (ecma_value_t this_arg, /**< this argument */
                                                               0,
                                                               JERRY_BITSINBYTE);
 
-        jerry_port_console ("\\u%02x%02x", byte_high, byte_low);
+        jerry_port_console ("\\u%" PRIu32 "02%" PRIu32 "02x", byte_high, byte_low);
```

Comment:
For the sake of precision: `byte_high` and `byte_low` _are_ of `uint32_t` type. However, basically, I start agreeing with you: it might be best keeping the "simple" format strings, using non-macro format specifiers and up-casting the arguments to int or unsigned int. (I guess, the existing format strings could work in most cases, perhaps some "%d"s might need a rewrite to "%u" to keep sign-correctness.)
@zherczeg Is this aligned with your ideas?